### PR TITLE
[SYCL][DeviceSanitizer] Add a test for demangle kernel name

### PR DESCRIPTION
**Pull request by yingcong-wu**
Add a test for demangle kernel name for UR PR https://github.com/oneapi-src/unified-runtime/pull/1355

### CODE REVIEW

File: sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 123456;
+  auto *array = sycl::malloc_device<double>(N, Q);
+
+  Q.submit([&](sycl::handler &h) {
+    h.parallel_for<class MyKernel>(
+        sycl::nd_range<1>(N + 1, 1),
+        [=](sycl::nd_item<1> item) { ++array[item.get_global_id(0)]; });
+  });
```
Comment (steffenlarsen) [inline sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp:15]: Do we really need to run such a big kernel just to test the output of the sanitizer?

```suggestion
  auto *data = sycl::malloc_device<double>(1, Q);

  Q.single_task<class MyKernel>(
        [=]() { ++data[0]; });
  });
```

File: sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux, cpu, aspect-fp64
+// RUN: %{build} %device_sanitizer_flags -O2 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t &> %t.txt ; FileCheck --input-file %t.txt %s
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 123456;
+  auto *array = sycl::malloc_device<double>(N, Q);
```
Comment (steffenlarsen) [inline sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp:9]: Remember to free this again. 😉

File: sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux, cpu, aspect-fp64
```
Comment (steffenlarsen) [inline sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp:1]: Is there a reason we use `double` for this test? Seems to me like we could just use `int` and avoid the additional requirement of `aspect-fp64`.

Comment (yingcong-wu) [reply sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp:1]: Good point.

File: sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp (hunk 1 above)
Comment (yingcong-wu) [reply sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp]: Using single_task cannot show that the kernel name is demangled as its name is quite a simple one that does not required demangle. So I reduce the range size instead.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by yingcong-wu** [comment]
Hi @intel/llvm-gatekeepers , could you please help merge this PR?

**Pull request merged**